src/modules/Visualizer/LayersTree/LayersTreeItem/WarningZoom/index.js

```diff
@@ -1,4 +1,17 @@
+import React from 'react';
+import processWarningAccordingToZoom from '../../../services/warningZoom';
 import WarningZoom from './WarningZoom';
 import { connectLayersTree } from '../../LayersTreeProvider/context';
 
-export default connectLayersTree('translate')(WarningZoom);
+const withWarningAccordingToZoom = WrappedComponent => props => {
+  const { map, layer, isActive, children } = props;
+  if (!isActive) return children;
+  const { showWarning, minZoomLayer } = processWarningAccordingToZoom(map, layer);
+  return (
+    <WrappedComponent display={showWarning} minZoomLayer={minZoomLayer} {...props}>
+      {children}
+    </WrappedComponent>
+  );
+};
+
+export default withWarningAccordingToZoom(connectLayersTree('translate')(WarningZoom));
```

Comment:
As soon as there is more than one HOC, it's a good idea to use a `compose()` function.

`compose()` may comes from a local helper like [the one used in terra-admin](https://github.com/Terralego/terra-admin/blob/master/src/utils/compose.js) or from any package as a dependency _(attention: all compose functions do not have the same signature. Sometimes arguments order and/or initial argument differ)_.

```suggestion
export default compose(
  withWarningAccordingToZoom,
  connectLayersTree('translate'),
)(WarningZoom);
```